x-pack/platform/packages/shared/kbn-streamlang/src/validation/validate_streamlang.test.ts

```diff
@@ -0,0 +1,1615 @@
+      expect(result.errors[0].message).toContain('processor #1 (set)');
+      expect(result.errors[0].message).not.toContain('my-custom-processor');
+    });
+  });
+
+  describe('empty DSL', () => {
+    it('should pass validation for empty steps array', () => {
+      const dsl: StreamlangDSL = {
+        steps: [],
+      };
+
+      const result = validateStreamlang(dsl, {
+        isWiredStream: true,
+        reservedFields: ['@timestamp'],
+      });
+
+      expect(result.isValid).toBe(true);
+      expect(result.errors).toHaveLength(0);
+    });
+  });
+
+  describe('keep fields validation', () => {
+    it('should allow special keep fields without namespacing', () => {
+      const dsl: StreamlangDSL = {
+        steps: [
+          {
+            action: 'set',
+            to: '@timestamp',
+            value: '2024-01-01',
+          },
+          {
+            action: 'set',
+            to: 'trace_id',
+            value: 'abc123',
+          },
+          {
+            action: 'set',
+            to: 'span_id',
+            value: 'def456',
+          },
+          {
+            action: 'set',
+            to: 'severity_text',
+            value: 'INFO',
+          },
+          {
+            action: 'set',
+            to: 'severity_number',
+            value: 9,
+          },
+          {
+            action: 'set',
+            to: 'event_name',
+            value: 'test.event',
+          },
+        ],
+      };
+
+      const result = validateStreamlang(dsl, { isWiredStream: true });
+
+      expect(result.isValid).toBe(true);
+      expect(result.errors).toHaveLength(0);
+    });
+
+    it('should allow body and body.text without namespacing', () => {
+      const dsl: StreamlangDSL = {
+        steps: [
+          {
+            action: 'set',
+            to: 'body',
+            value: { text: 'message' },
+          },
+          {
+            action: 'set',
+            to: 'body.text',
+            value: 'message',
+          },
+        ],
+      };
+
+      const result = validateStreamlang(dsl, { isWiredStream: true });
+
+      expect(result.isValid).toBe(true);
+      expect(result.errors).toHaveLength(0);
+    });
+
+    it('should allow observed_timestamp and other keep fields', () => {
+      const dsl: StreamlangDSL = {
+        steps: [
+          {
+            action: 'set',
+            to: 'observed_timestamp',
+            value: '2024-01-01',
+          },
+          {
+            action: 'set',
+            to: 'dropped_attributes_count',
+            value: 0,
+          },
+          {
+            action: 'set',
+            to: 'scope',
+            value: { name: 'test' },
+          },
+        ],
+      };
+
+      const result = validateStreamlang(dsl, { isWiredStream: true });
+
+      expect(result.isValid).toBe(true);
+      expect(result.errors).toHaveLength(0);
+    });
+
+    it('should still reject truly non-namespaced custom fields', () => {
```

Comment:
nit: the test is samea as ^ `"should fail validation when fields are not namespaced in wired streams"`